src/gridfs.rs

```diff
@@ -4,49 +4,66 @@
 pub mod options;
 
 use core::task::{Context, Poll};
-use std::pin::Pin;
+use std::{pin::Pin, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+use tokio::io::{AsyncWriteExt, ReadBuf};
+use tokio_util::compat::FuturesAsyncWriteCompatExt;
 
 use crate::{
+    bson::{doc, oid::ObjectId, Binary, Bson, DateTime, Document},
     concern::{ReadConcern, WriteConcern},
     cursor::Cursor,
-    error::Result,
+    error::{ErrorKind, Result},
+    options::{FindOneOptions, FindOptions},
     selection_criteria::SelectionCriteria,
+    Collection,
     Database,
 };
-use bson::{oid::ObjectId, Bson, DateTime, Document};
+
 use options::*;
-use serde::{Deserialize, Serialize};
-use tokio::io::ReadBuf;
 
 pub const DEFAULT_BUCKET_NAME: &str = "fs";
 pub const DEFAULT_CHUNK_SIZE_BYTES: u32 = 255 * 1024;
 
 // Contained in a "chunks" collection for each user file
-struct Chunk {
-    id: ObjectId,
-    files_id: Bson,
-    n: u32,
+#[derive(Debug, Deserialize, Serialize)]
+pub struct Chunk {
```

Comment:
going to disable the lint suppression for missing docs in places like these and add them in when I do RUST-1395